docs/source/deployment/airflow.md

```diff
@@ -0,0 +1,181 @@
+# Apache Airflow
+
+Apache Airflow is a popular open-source workflow management platform. It is a suitable engine to orchestrate and execute a pipeline authored with Kedro because workflows in Airflow are modelled and organised as [DAGs](https://en.wikipedia.org/wiki/Directed_acyclic_graph).
```

Comment:
Note for the future: we should have an entry for DAGs in our Glossary! https://docs.kedro.org/en/stable/resources/glossary.html